### PR TITLE
Get the content type of an embedded entry in a self link

### PR DESCRIPTION
**Pull request by asooben**
Retrieve the content type of a payload in a self link

### CODE REVIEW

**Comment by mtemenos** [comment]
What's the use case for this?

**Comment by asooben** [comment]
Self link with an embedded payload.
Self links do not have any content type attribute and the content type is required in the integration testing framework to determine which payload handler (atom, text, etc...) to use.

**Comment by mtemenos** [comment]
So we're changing the framework to make an integration test pass as opposed to satisfying an actual functional requirement. Does that seem right to you?

**Comment by aphethean** [comment]
Agree with Martins point.

Should the Content-Type header of the response contain the information you need.  Self link is after all the identifier for the current response.

I also hope you are not using the content type attribute on other links in anything other than an advisory manner.  You should always rely on the response you actually receive.

Aaron

File: useragent-generic-java/src/main/java/com/temenos/useragent/generic/mediatype/AtomLinkHandler.java [hunk 1]
```diff
@@ -105,19 +108,21 @@ private void buildEmbeddedPayload() {
...
-			//As we lose the reference to the parent after the conversion to String, we need to explicitly set the baseUri
-			entryElement.setBaseUri(abderaLink.getBaseUri());
 			if (entryElement != null) {
+				//As we lose the reference to the parent after the conversion to String, we need to explicitly set the baseUri
+				entryElement.setBaseUri(abderaLink.getBaseUri());				
 				content = getContent(entryElement);
 			}
 		}
+		
+		String contentType = abderaLink.getAttributeValue("type")!=null?
```
Comment (mtemenos) [inline useragent-generic-java/src/main/java/com/temenos/useragent/generic/mediatype/AtomLinkHandler.java]: As discussed please rely on the response content type instead.

**Comment by asooben** [comment]
This change is an update to the testing framework to pick out content type when it's not available on a link. It is not setting content-type on self links nor is it making any change to IRIS core.

Point taken, the testing framework will be updated accordingly at a later stage.

**Pull request closed without merging**